sdk/python/kfp/_client.py

```diff
@@ -99,15 +99,36 @@ def list_experiments(self, page_token='', page_size=10, sort_by=''):
         page_token=page_token, page_size=page_size, sort_by=sort_by)
     return response
 
-  def get_experiment(self, experiment_id):
+  def get_experiment(self, experiment_id=None, experiment_name=None):
     """Get details of an experiment
+    Either experiment_id or experiment_name is required
     Args:
-      id of the experiment.
+      experiment_id: id of the experiment. (Optional)
+      experiment_name: name of the experiment. (Optional)
     Returns:
       A response object including details of a experiment.
     Throws:
-      Exception if experiment is not found.        
+      Exception if experiment is not found or None of the arguments is provided
     """
+    if experiment_id is None and experiment_name is None:
+      raise ValueError('Either experiment_id or experiment_name is required')
+    if experiment_id is not None:
+      return self._experiment_api.get_experiment(id=experiment_id)
+    list_experiments_response = self.list_experiments(page_size=100)
```

Comment:
Can be simplified:

```
next_page_token=''
while next_page_token is not None:
    list_experiments_response = self.list_experiments(page_size=100, page_token=next_page_token)
    next_page_token = list_experiments_response.next_page_token
    for experiment in list_experiments_response.experiments:
        if experiment.name == experiment_name:
            return self._experiment_api.get_experiment(id=experiment_id)
raise ValueError('No experiment is found with name %s.'.format(experiment_name))
```

Reply:
does list_experiment works with empty next_page_token?

Reply:
Default value is '', and it works.

Reply:
thanks for pointing that out

Reply:
Done.